src/main/resources/webinterface/js/requests.js

```diff
@@ -0,0 +1,280 @@
+    if (indexRefreshInterval === undefined) {
+        indexRefresh();
+        indexRefreshInterval = setInterval(indexRefresh, Constants.Intervals.INDEX_REFRESH);
+    }
+}
+
+/**
+ * Stop the index refresh requests.
+ */
+function stopIndexRefresh() {
+    if (indexRefreshInterval !== undefined) {
+        clearInterval(indexRefreshInterval);
+        indexRefreshInterval = undefined;
+    }
+}
+
+/**
+ * Start the status update requests.
+ */
+function startStatusUpdates() {
+    if (supportsWebSockets === false) {
+        if (statusUpdateInterval === undefined) {
+            requestStatusUpdate();
+            statusUpdateInterval = setInterval(requestStatusUpdate, Constants.Intervals.STATUS_UPDATE);
+        }
+    } else if (supportsWebSockets === true && !isSocketOpen()) {
+        createWebSocket();
+    }
+}
+
+/**
+ * Stop the status update requests.
+ */
+function stopStatusUpdates() {
+    if (statusUpdateInterval !== undefined) {
+        clearInterval(statusUpdateInterval);
+        statusUpdateInterval = undefined;
+    } else if (isSocketOpen()) {
+        closeWebSocket();
+    }
+}
+
+// ================================================================================================
+// =========================================== REQUESTS ===========================================
+// ================================================================================================
+
+/**
+ * Requests an index refresh.
+ *
+ * The given callback is called with either true, false, or a GameState, to indicate what the server replied.
+ * true = you can still join.
+ * false = you cannot join / error occurred.
+ * GameState = you are already in the game, and should go to the given state view.
+ *
+ * The callback WILL always be called.
+ *
+ * @param callback
+ *      function that will be called when the server has responded
+ */
+function requestIndexRefresh(callback) {
+    $.ajax({
+        type: "POST",
+        url: "/indexrefresh",
+        success: function(data) {
+            if (data === "y") {
+                callback(true);
+            } else if (data === "n") {
+                callback(false);
+            } else {
+                callback(GameStates[data]);
+            }
+        },
+        error: function(jqxhr, status, error) {
+            callback(false);
+            handleIndexRefreshError(jqxhr, status, error);
+        }
+    });
+};
+
+/**
+ * Requests the server to join the game.
+ * 
+ * On success, this function calls handleJoinGame.
+ * On failure, this function calls handleGlobalRequestError.
+ */
+function requestJoinGame() {
+    $.ajax({
+        type: "POST",
+        url: "/login",
+        success: handleJoinGame,
+        error: handleGlobalRequestError
+    });
+}
+
+/**
+ * Requests the server to set the team, either via websocket or normally.
+ *
+ * @param team
+ *      the team to set
+ */
+function requestSetTeam(team) {
+    requestNormal_setTeam(team);
+    
+    //TODO WebSocket
```

Comment:
The websocket implementation of setTeam is not done. We might not need it. Testing must show.